packages/react/src/components/ComposedModal/next/ModalFooter.js

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../Button';
+import ButtonSet from '../../ButtonSet';
+import classNames from 'classnames';
+import { usePrefix } from '../../../internal/usePrefix';
+
+export const ModalFooter = React.forwardRef(function ModalFooter(
+  {
+    className,
+    primaryClassName,
+    secondaryButtons,
+    secondaryClassName,
+    secondaryButtonText,
+    primaryButtonText,
+    primaryButtonDisabled,
+    closeModal, // eslint-disable-line
+    onRequestClose, // eslint-disable-line
+    onRequestSubmit, // eslint-disable-line
+    children,
+    danger,
+    inputref,
+    ...other
+  },
+  ref
+) {
+  const prefix = usePrefix();
+
+  function handleRequestClose(evt) {
+    closeModal(evt);
+    onRequestClose(evt);
+  }
+
+  const footerClass = classNames({
+    [`${prefix}--modal-footer`]: true,
+    [className]: className,
+    [`${prefix}--modal-footer--three-button`]:
+      Array.isArray(secondaryButtons) && secondaryButtons.length === 2,
+  });
+
+  const primaryClass = classNames({
+    [primaryClassName]: primaryClassName,
+  });
+
+  const secondaryClass = classNames({
+    [secondaryClassName]: secondaryClassName,
+  });
+
+  const SecondaryButtonSet = () => {
+    if (Array.isArray(secondaryButtons) && secondaryButtons.length <= 2) {
+      return secondaryButtons.map(
+        ({ buttonText, onClick: onButtonClick }, i) => (
+          <Button
+            key={`${buttonText}-${i}`}
+            className={secondaryClass}
+            kind="secondary"
+            onClick={onButtonClick || handleRequestClose}>
+            {buttonText}
+          </Button>
+        )
+      );
+    }
+    if (secondaryButtonText) {
+      return (
+        <Button
+          className={secondaryClass}
+          onClick={handleRequestClose}
+          kind="secondary">
+          {secondaryButtonText}
+        </Button>
+      );
+    }
+    return null;
+  };
```

Comment:
Since this is being used as a component, could we pull it out as a local component and pass in the data it's using from scope as props?